### PR TITLE
Cloud parser refactor

### LINKED ISSUE #73: Modify parser code flow to support cloud deployment
## Problem
We want to be able to deploy uploader modules to the cloud, but file system walking stops us, as you cannot treat blob storage as a standard file system.

The problem stems from `parsers/<parser>/sample_parser.py#_parse_sample_list` having calls to `os.path` and `os.walk`

We do this because the `_parse_sample_list` function is only given the samplesheet file, and then figures out which files are which samples from there, making walking the filesystem necessary. 

## Proposed Solutions
Solution A:
---
At the point in code where the walk is called, we have another module take control, and then either do the walk as normal, or walk the cloud structure, and return the file list.

This would be rather straight forward, but requires us to write a cloud environment specific piece of code.

Code flow:

Cloud app detects new run
- gets sample sheet
- initializes parser with cloud credentials
- starts parsing
  - detects cloud is set up
  - parses cloud structure
    - error return would need to be updated to handle cloud uri's
  - completes parsing,
  - returns run object with cloud uri attached (don't know how this step would work yet)
- cloud app moves to upload

Solution B:
---
Add 2 new methods to the parsers, `get_data_directory` which returns the parser specific folder structure, and `build_data_directory_file_list(data_directory)`, and then modify the parsers to accept the data directory file list, instead of building itself in the method that builds the run.

This would would make the parser handle the file structure information, as well as the validation layer,
This moves the burden of translating the storage structure of azure or another cloud system to the cloud app.

Code Flow:

Cloud app detects new run
- gets sample sheet
- asks parser for data directory (this can be different per parser)
- cloud app creates list of data files in data directory, (keeps this for later)
- starts parsing (passing in the data file list)
  - doesn't need to know about cloud structure, so just returns the run object
  - on error return, only returns file name, no change needed
- cloud app moves to upload
  - on upload, correlates the sequence file, to the uri (they have identical names so it's simple)

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## Description of changes
Restructured some parser code to support deployment in cloud environments.
See solution B in #73 

## Related issue
Fixes #73 

## Checklist

* [x] CHANGELOG.md updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: scripts/update_version.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Use example:
```
[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply scripts/update_version.sh:6]: Added.

File: iridauploader/parsers/common.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import os
```
[the remark this answers is not in the document]
Comment (JeffreyThiessen) [reply iridauploader/parsers/common.py:8]: Added.

File: iridauploader/parsers/common.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+"""
+This file has generic utility methods that can be used by all parsers
+
+These methods can rely on the os module to function, and therefor not to be used with cloud environments.
```
Comment (tom114) [inline iridauploader/parsers/common.py:4]: therefor->therefore

Comment (JeffreyThiessen) [reply iridauploader/parsers/common.py]: oops, fixed

**Pull request merged**